src/main/java/com/openshift/internal/restclient/DefaultClient.java

```diff
@@ -249,12 +249,13 @@ public  boolean supports(Class<? extends ICapability> capability) {
 	
 	@SuppressWarnings("unchecked")
 	@Override
-	public <T extends ICapability> void accept(CapabilityVisitor<T> visitor){
+	public <T extends ICapability, R> R accept(CapabilityVisitor<T, R> visitor, R unsupportedCapabililityValue){
```

Comment:
@adietish needed the ability to return a value from the visitor, but also handle the case where the visitor wasn't 'visited'.  Without providing a 'default' value you can run into either null pointers our class cast exceptions.  It's not as pretty as it used to be :(